core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -251,11 +274,122 @@ public void testAddSortOrderFromJson() {
     assertEquals(action, expected, MetadataUpdateParser.fromJson(json));
   }
 
+  @Test
+  public void testSetDefaultSortOrderToJson() {
+    String action = MetadataUpdateParser.SET_DEFAULT_SORT_ORDER;
+    int sortOrderId = 2;
+    String expected = String.format("{\"action\":\"%s\",\"sort-order-id\":%d}", action, sortOrderId);
+    MetadataUpdate update = new MetadataUpdate.SetDefaultSortOrder(sortOrderId);
+    String actual = MetadataUpdateParser.toJson(update);
+    Assert.assertEquals("Set default sort order should serialize to the correct JSON value", expected, actual);
+  }
+
+  @Test
+  public void testSetDefaultSortOrderFromJson() {
+    String action = MetadataUpdateParser.SET_DEFAULT_SORT_ORDER;
+    int sortOrderId = 2;
+    String json = String.format("{\"action\":\"%s\",\"sort-order-id\":%d}", action, sortOrderId);
+    MetadataUpdate expected = new MetadataUpdate.SetDefaultSortOrder(sortOrderId);
+    assertEquals(action, expected, MetadataUpdateParser.fromJson(json));
+  }
+
+  // TODO - AddSnapshot tests.
+
+  // TODO - Tests that handle multiple snapshot ids.
+  @Test
+  public void testRemoveSnapshotsFromJson() {
```

Comment:
Currently, the spec allows for `RemoveSnapshots` to have multiple snapshots. But the code only allows for one.

Should we implement in another PR, or move forward with just 1 allowed temporarily?

Reply:
Move forward with just 1 allowed, temporarily. I think in a follow up PR, we should support multiple snapshots in `RemoveSnapshot` (renamed to `RemoveSnapshots`).

Reply:
Yeah I moved forward with just 1 allowed. Would you like me to change the class name to be singular for now? Currently the singular constructor class is `RemovePartitions`.

We can either accept one value and then collect it into a set and expose two constructors or make two classes.